plugin.py

```diff
@@ -54,6 +59,27 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         self._api = None  # type: Optional[ApiWrapperInterface]
         super().__init__(*args, **kwargs)
 
+    @classmethod
+    def is_allowed_to_start(
+        cls,
+        window: sublime.Window,
+        initiating_view: Optional[sublime.View] = None,
+        workspace_folders: Optional[List[WorkspaceFolder]] = None,
+        configuration: Optional[ClientConfig] = None
+    ) -> Optional[str]:
+        if workspace_folders:
+            cls.register_rename_import_file_watcher(cls, window, workspace_folders[0].path)
+        return None
+
+    def register_rename_import_file_watcher(self, window: sublime.Window, root_path: str) -> Optional[FileWatcher]:
+        settings = sublime.load_settings("LSP-typescript.sublime-settings")
+        update_imports_on_file_move_setting = settings.get('settings', {}).get('updateImportsOnFileMove', "prompt")
+        if update_imports_on_file_move_setting == "never":
+            return
+        file_watcher = get_file_watcher_implementation()
+        if file_watcher:
+            self.rename_import_handler = HandleRenameImport(window, root_path, file_watcher)
```

Comment:
You are calling this method from a class method so `self` is in fact a class object (`cls`) and not an instance. You can't just set it on a class object as that will overwrite previously created instance.

Seems like the best (only) place would be `__init__` and getting the workspace folders from `Session`.